data/crac-creation/crac-creator-csa-profiles/src/main/java/com/farao_community/farao/data/crac_creation/creator/csa_profile/crac_creator/remedial_action/CsaProfileRemedialActionsCreator.java

```diff
@@ -159,10 +158,26 @@ private void addOnContingencyStateUsageRules(RemedialActionAdder<?> remedialActi
                 .withUsageMethod(usageMethod).add());
     }
 
-    private void checkAllContingenciesLinkedToRaHaveTheSameConstraintKind(String remedialActionId, Set<PropertyBag> linkedContingencyWithRAs, String firstKind) {
+    private void checkElementCombinationConstraintKindsCoherence(String remedialActionId, Set<PropertyBag> linkedContingencyWithRAs) {
```

Comment:
i suggest a change a signature, to have the whole map and get the set of contingencies related to remedial action inside this method to make it clearer

Reply:
The `linkedContingencyWithRAs` property bag object is already created in the method `createAndAddRemedialActions` which calls `checkElementCombinationConstraintKindsCoherence`. This would duplicate the use of `CsaProfileCracUtils.getMappedPropertyBagsSet(contingencyWithRemedialActionsPropertyBags, CsaProfileConstants.GRID_STATE_ALTERATION_REMEDIAL_ACTION);`. What do you think?